### ISSUE TITLE
Why loss is nan in training

### ISSUE THREAD

**Issue by yijingru** [amdegroot/ssd.pytorch#73]
Hi, I was trying to train this model but the loss was nan. Do you know where is wrong?

**Comment by yijingru** [comment]
I found the bug is in encode function. When 'matched ' is all zero, the encoded box coords will become -inf or nan

**Comment by amdegroot** [comment]
What is your batch size?

**Comment by yijingru** [comment]
Thank you for relying. I solved the problem. The nan is caused by the division operation '/'.

**Comment by tianyu-tristan** [comment]
@yijingru hi could you pin point the line of code that cause the problem? I'm experiencing the same, thanks a lot!

**Comment by yijingru** [comment]
@tianyu-tristan Hi, I think it's caused by the version of python? I just modify several '/ 'operations like in voc0702.py, when normalize the groundtruth bounding box, I changed cur_pt/width to float(cur_pt) / width so that the coordinates will not be 0.

**Comment by tianyu-tristan** [comment]
@yijingru thanks, i see what you mean. I may have messed up somewhere else since I'm on python3. thanks anyway!

**Comment by eng100200** [comment]
@yijingru divsion '/' where in multibox loss??/

**Comment by pxkc** [comment]
> How could you figure it out, I met the same problem using all default values. I mean, do I need to change batch_size or learning rate?

can you give me some advice?  Thank you.

**Comment by usherbob** [comment]
> > How could you figure it out, I met the same problem using all default values. I mean, do I need to change batch_size or learning rate?
> 
> can you give me some advice? Thank you.

I set batch_size 32 and learning rate 4e-3, you could try it.

**Comment by creabob** [comment]
in voc0712.py change code as
cur_pt = cur_pt / width if i % 2 == 0 else cur_pt / height
==>
cur_pt = 1.0 * cur_pt / width if i % 2 == 0 else 1.0 * cur_pt / height

**Comment by jaehobang** [comment]
> @yijingru thanks, i see what you mean. I may have messed up somewhere else since I'm on python3. thanks anyway!

@tianyu-tristan Did you find out what was the fix to the problem? I am trying this out on a different dataset, and I am encountering this problem. I am using Python 3+, so division shouldn't be the issue.
